test/ejabberd_tests/tests/rest_SUITE.erl

```diff
@@ -183,6 +185,53 @@ messages_are_archived(Config) ->
         BobJID = maps:get(sender, Previous)
     end).
 
+send_and_wait(AliceJID, BobJID, Msg) ->
+    M = #{caller => BobJID, to => AliceJID, msg => Msg},
+    {?OK, _} = post(<<"/messages">>, M),
+    M1 = #{caller => AliceJID, to => BobJID, msg => Msg},
+    {?OK, _} = post(<<"/messages">>, M1),
+    timer:sleep(2000).
```

Comment:
How about checking every 1000ms but no longer than 10s f.e.
BTW, why timestamp in sec and not in milisec?